docs/oauth-setup.md

```diff
@@ -0,0 +1,143 @@
+- The other fields can contain any information that you'd like, as this client will be specific to your instance of SHIELD
+- Github will now generate a client id and client secret that you will need to copy to the config yaml file you feed to SHIELD.
+
+### Add the configuration for your Github OAuth client
+
+This will be added to the already existing configuration YAML file you use to deploy shield under a new field called `auth:`
+
+#### required fields are:
+
+- name - this is the name that will be displayed in the UI and CLI when your users are interacting with SHIELD 
+- identifier - this is the field that you put in the "Authorization callback URL" field when you set up your Github OAuth client. The SHIELD backend will use this to identify which OAuth Client was in use for things like logs and auditing.
+- backend - this is a static value defined by SHIELD, and will always be `github` for this configuration
+- properties:
+  - client_id - this was auto generated by Github when you created your OAuth client
+  - client_secret - this was also auto generated by Github when you created your OAuth client
+  - mapping - this section will contain the mapping from github orgs to SHIELD tenants.  For each Github organization you may configure a mapping from a SHIELD tenant (that you make up the name of) to a github org name.  Then, underneath that,  you may define, based on the Github teams within the above Github org, what Role a given user will have in SHIELD.  You may also define a base case role that all users will get if they do not belong to one of the matching Github teams that you have provided.
+    - <github_tenant_name> - this can be any name you would like.  SHIELD will create an internal SHIELD tenant with this given name
+    - rights - this section will contain the mappings from Github team to a predefined SHIELD role, as wel as a default role for all users that don't belong to any of the specified Github teams
+      - the shield roles are as follows:
+        - admin
+        - operator
+        - engineer
+
+#### Here is an example config used for SHIELD development:
+
+```yaml
+auth:
+  - name:       Github
+    identifier: public-gh
+    backend:    github
+    properties:
+      client_id:     435abbcea49c948f8ab3
+      client_secret: ffab373bc287d40ba73476bff3f449
+      mapping:
+        starkandwayne:           # <-- github org name
+          tenant: starkandwayne  # <-- shield tenant name
+          rights:
+            - team: Owners       # <-- github team name
+              role: admin        # <-- shield role name
+            - team: Engineering  #   (first match wins)
+              role: engineer
+            - role: operator     # = (default match)
+        cloudfoundry-community:
+          tenant: CF Community
+          rights:
+            - role: engineer
+
+  - name:       UAA
+  ...
+```
+
+## UAA
+
+This configuration gives you a single admin user with the password "PASSWORD"
```

Comment:
We should probably remove this, IIRC it was from the dev version for spinning a test UAA.

Instead, we should explain why the operator would want to use UAA, in what circumstances.  What are the capabilities and limitations?  What scenarios does it make sense to use the UAA provider?